farming/logics/traits/data.rs

```diff
@@ -0,0 +1,54 @@
+use ink_prelude::vec::Vec;
+use ink_storage::{
+    traits::*,
+    Mapping,
+};
+use openbrush::traits::AccountId;
+use scale::{
+    Decode,
+    Encode,
+};
+
+pub const STORAGE_KEY: u32 = openbrush::storage_unique_key!(Data);
+
+#[derive(Encode, Decode, SpreadLayout, PackedLayout, SpreadAllocate, Default)]
+#[cfg_attr(
+    feature = "std",
+    derive(Debug, PartialEq, Eq, scale_info::TypeInfo, StorageLayout)
+)]
+pub struct Pool {
+    pub acc_arsw_per_share: u128,
+    pub last_reward_block: u32,
```

Comment:
Isn’t it better to use BlockNumber for the type of it?

Reply:
If you think this is more readable yes.
but then all u128 should be Balance

Reply:
Yep, think it is more readable. Let's do it as a separate PR, since then we should change other projects' typing, not alone this one.